jerry-ext/args/args.c

```diff
@@ -0,0 +1,85 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "args-internal.h"
+#include "jerryscript-ext/args.h"
+#include "jerryscript.h"
+
+/**
+ * Validate the JS arguments and assign them to the native arguments.
+ *
+ * @return jerry undefined: the validator passes,
+ *         jerry error: the validator fails.
+ */
+jerry_value_t
+jerryx_arg_transform_args (const jerry_value_t *js_arg_p, /**< points to the js argument */
+                           const jerry_length_t js_arg_cnt, /**< the count of the js arguments */
+                           const jerryx_arg_t *c_arg_p, /**< points to the jerryx_arg_t */
+                           jerry_length_t c_arg_cnt) /**< the count of the native argument */
+{
+  jerry_value_t ret = jerry_create_undefined ();
+
+  jerryx_arg_js_iterator_t iterator =
+  {
+    .js_arg_p = js_arg_p,
+    .js_arg_cnt = js_arg_cnt,
+    .js_arg_idx = 0
+  };
+
+  for (; c_arg_cnt != 0 && !jerry_value_has_error_flag (ret); c_arg_cnt--, c_arg_p++)
+  {
+    ret = c_arg_p->func (&iterator, c_arg_p);
+  }
+
+  return ret;
+} /* jerryx_arg_transform_args */
+
+/**
+ * Validate the this value and the JS arguments,
+ * and assign them to the native arguments.
+ *
+ * @return jerry undefined: the validator passes,
+ *         jerry error: the validator fails.
+ */
+jerry_value_t
+jerryx_arg_transform_args_with_this (const jerry_value_t this_val, /**< the this_val for the external function */
```

Comment:
Question: it seems the first argument is ignored in this text, which is probably the `this`, doesn't it? What is the point of having a variant without `this` when it can be skipped easily?